rclcpp/include/rclcpp/node.hpp

```diff
@@ -67,22 +67,29 @@ class Node : public std::enable_shared_from_this<Node>
   /// Create a new node with the specified name.
   /**
    * \param[in] node_name Name of the node.
+   * \param[in] name_space Namespace of the node.
    * \param[in] use_intra_process_comms True to use the optimized intra-process communication
    * pipeline to pass messages between nodes in the same process using shared memory.
    */
   RCLCPP_PUBLIC
-  explicit Node(const std::string & node_name, bool use_intra_process_comms = false);
+  explicit Node(
+    const std::string & node_name,
+    const std::string & name_space = "",
```

Comment:
do you have a specific reason for passing the name_space explicitly and don't capture it in a struct aka `node_options`?

we could also think of putting `use_intra_process_comms` into it.